src/transformers/pipelines/text_to_audio.py

```diff
@@ -0,0 +1,134 @@
+from typing import List, Union
+
+import numpy as np
+
+from transformers import Pipeline, SpeechT5HifiGan
+
+from ..utils import is_torch_available
+
+
+if is_torch_available():
+    from ..models.auto.modeling_auto import MODEL_FOR_TEXT_TO_SPECTROGRAM_MAPPING
+
+DEFAULT_VOCODER_ID = "microsoft/speecht5_hifigan"
+
+
+class TextToAudioPipeline(Pipeline):
+    """
+    Text-to-audio generation pipeline using any `AutoModelForTextToWaveform` or `AutoModelForTextToSpectrogram`. This
+    pipeline generates an audio file from an input text and optional other conditional inputs.
+
+    Example:
+
+    ```python
+    >>> from transformers import pipeline
+
+    >>> classifier = pipeline(model="suno/bark")
+    >>> audio = pipeline("Hey it's HuggingFace on the phone!", speaker_embeddings="v2/en_speaker_1")
+    ```
+
+    Learn more about the basics of using a pipeline in the [pipeline tutorial](../pipeline_tutorial)
+
+
+    This pipeline can currently be loaded from [`pipeline`] using the following task identifiers: `"text-to-speech"` or
+    `"text-to-audio"`.
+
+    See the list of available models on [huggingface.co/models](https://huggingface.co/models?filter=text-to-speech).
+    """
+
+    def __init__(self, *args, vocoder=None, sampling_rate=None, **kwargs):
+        super().__init__(*args, **kwargs)
+
+        if self.framework == "tf":
+            raise ValueError("The TextToAudioPipeline is only available in PyTorch.")
+
+        use_forward_bool = "PreTrainedModel" in str(self.model.generate)
+        self.forward_method = self.model if use_forward_bool else self.model.generate
```

Comment:
There has been a change in how to decide if we use forward or generate.

Reply:
I don't think this method generalises, e.g. if I try it for Whisper, an auto-regressive ASR model, I get:
```python
from transformers import WhisperForConditionalGeneration, WhisperConfig

config = WhisperConfig()
model = WhisperForConditionalGeneration(config)

print("PreTrainedModel" in str(model.generate))
```
**Print Output:**
```output
False
```

What about just checking `model.can_generate()`?

Reply:
It seems that it is the expected behavior right ? (correct me if I'm wrong!)

Because `use_forward_bool=False` for `WhisperForConditionalGeneration`, we would have the following line that would be `self.model.generate`:
```self.forward_method = self.model if use_forward_bool else self.model.generate```

Reply:
Aha yes you're right! Got them the wrong way round - marking as resolved!

Reply:
Actually still don't think this generalises - e.g. for the CLAP model (which is non auto-regressive):
```python
from transformers import ClapTextModelWithProjection, ClapTextConfig

config = ClapTextConfig()
model = ClapTextModelWithProjection(config)

print("PreTrainedModel" in str(model.generate))
```
**Print Output:**
```output
False
```
=> this should return `True` no?

Reply:
You are very right, it should actually be `GenerationMixin` instead of `PreTrainedModel`.
The thing is that can_generate don't work with every models (exemple: `SpeechT5ForTextToSpeech`)

Reply:
Works as expected with models that can generate and models that can't:

```python
from transformers import WhisperForConditionalGeneration, WhisperConfig

config = WhisperConfig()
model = WhisperForConditionalGeneration(config)


forward_method = model if "PreTrainedModel" in str(model.generate) else model.generate
print("GenerationMixin" in str(model.generate))


from transformers import ClapTextModelWithProjection, ClapTextConfig

config = ClapTextConfig()
model = ClapTextModelWithProjection(config)

print("GenerationMixin" in str(model.generate))



from transformers import BertForTokenClassification, BertConfig

config = BertConfig()
model = BertForTokenClassification(config)

print("GenerationMixin" in str(model.generate))


from transformers import SpeechT5ForSpeechToSpeech, SpeechT5Config

config = SpeechT5Config()
model = SpeechT5ForSpeechToSpeech(config)

print("GenerationMixin" in str(model.generate))

from transformers import SpeechT5ForTextToSpeech, SpeechT5Config

config = SpeechT5Config()
model = SpeechT5ForTextToSpeech(config)

print("GenerationMixin" in str(model.generate))
```

Reply:
Would love your feedback @sanchit-gandhi

Reply:
Hmm `PreTrainedModel` inherits from `GenerationMixin`: https://github.com/huggingface/transformers/blob/0ebe7ae16076f727ac40c47f8f9167013c4596d8/src/transformers/modeling_utils.py#L1029

So any model that inherits from `PreTrainedModel` will also have `GenerationMixin`

Would it not be easiest to set `can_generate` to `True` for `SpeechT5ForTextToSpeech` in a separate PR and then always check `can_generate`?

Reply:
I guess it would be possible, yes!

Reply:
Following this discussion, #25493 add `can_generate=True` for `SpeechT5ForTextToSpeech`. I've also change the way we test if a model can generate !